rental_pricelist/models/product.py

```diff
@@ -342,39 +353,45 @@ def write(self, vals):
-            ext_vals["rental_of_day"] = True
-            ext_vals["rental_price_day"] = vals.get("rental_price_day")
-            del vals["rental_of_day"]
-            if "rental_price_day" in vals:
-                del vals["rental_price_day"]
-        if vals.get("rental_of_hour", False):
-            ext_vals["rental_of_hour"] = True
-            ext_vals["rental_price_hour"] = vals.get("rental_price_hour")
-            del vals["rental_of_hour"]
-            if "rental_price_hour" in vals:
-                del vals["rental_price_hour"]
-        res = super().create(vals)
-        if "income_analytic_account_id" in vals:
-            ext_vals["income_analytic_account_id"] = vals.get(
-                "income_analytic_account_id", False
-            )
-        else:
-            ext_vals["income_analytic_account_id"] = res.income_analytic_account_id.id
-        if "expense_analytic_account_id" in vals:
-            ext_vals["expense_analytic_account_id"] = vals.get(
-                "expense_analytic_account_id", False
-            )
-        else:
-            ext_vals["expense_analytic_account_id"] = res.expense_analytic_account_id.id
+        for vals in vals_list:
+            if vals.get("rental_of_month", False):
+                ext_vals["rental_of_month"] = True
+                ext_vals["rental_price_month"] = vals.get("rental_price_month")
+                del vals["rental_of_month"]
+                if "rental_price_month" in vals:
+                    del vals["rental_price_month"]
+            if vals.get("rental_of_day", False):
+                ext_vals["rental_of_day"] = True
+                ext_vals["rental_price_day"] = vals.get("rental_price_day")
+                del vals["rental_of_day"]
+                if "rental_price_day" in vals:
+                    del vals["rental_price_day"]
+            if vals.get("rental_of_hour", False):
+                ext_vals["rental_of_hour"] = True
+                ext_vals["rental_price_hour"] = vals.get("rental_price_hour")
+                del vals["rental_of_hour"]
+                if "rental_price_hour" in vals:
+                    del vals["rental_price_hour"]
+        res = super().create(vals_list)
+        for vals in vals_list:
+            if "income_analytic_account_id" in vals:
+                ext_vals["income_analytic_account_id"] = vals.get(
+                    "income_analytic_account_id", False
+                )
+            else:
+                ext_vals[
+                    "income_analytic_account_id"
+                ] = res.income_analytic_account_id.id
+            if "expense_analytic_account_id" in vals:
+                ext_vals["expense_analytic_account_id"] = vals.get(
+                    "expense_analytic_account_id", False
+                )
+            else:
+                ext_vals[
+                    "expense_analytic_account_id"
+                ] = res.expense_analytic_account_id.id
         res.write(ext_vals)
```

Comment:
ext_vals contains only the value of one created product. It should not be written in all the new created products.